lib/haml/railtie.rb

```diff
@@ -10,13 +10,32 @@
 module Haml
   class Railtie < ::Rails::Railtie
     initializer :haml do |app|
-      require "haml/template"
-      if defined?(::Sass::Rails::SassTemplate) && app.config.assets.enabled
-        require "haml/sass_rails_filter"
+      if defined?(ActiveSupport)
+        ActiveSupport.on_load(:before_initialize) do
+          ActiveSupport.on_load(:action_view) do
```

Comment:
> fairly robust (conservative) 

Being too conservative makes hard to understand why it is implemented as such. Since you have a reproductive repo, you can confirm which is unnecessary. Please check it and remove it.